client/components/tinymce/plugins/mentions/suggestion.jsx

```diff
@@ -3,74 +3,67 @@
  */
 import React from 'react';
 
-const Suggestion = React.createClass( {
-	displayName: 'Suggestion',
-
-	propTypes: {
-		avatarUrl: React.PropTypes.string,
-		username: React.PropTypes.string,
-		fullName: React.PropTypes.string,
-		query: React.PropTypes.string
-	},
-
-	getDefaultProps() {
-		return {
-			avatarUrl: '',
-			username: '',
-			fullName: '',
-			query: ''
-		};
-	},
-
-	getRegExpFor( type, textToHighlight ) {
-		const expressions = {};
-
-		expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
-		expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
-
-		return new RegExp( expressions[ type ], 'ig' );
-	},
-
-	highlight( content, textToHighlight, type ) {
-		const matcher = this.getRegExpFor( type, textToHighlight ),
-			matches = matcher.exec( content );
-
-		if ( matches ) {
-			const highlights = [];
-			let highlighted = false;
-
-			for ( let i = 1, length = matches.length; i < length; i++ ) {
-				let item = matches[ i ];
-
-				if ( textToHighlight.toLowerCase() === item.toLowerCase() && ! highlighted ) {
-					item = <strong className="mentions__highlight" key={ i }>{ matches[ i ] }</strong>;
-					highlighted = true;
-				}
-
-				highlights.push( item );
-			}
+const getRegExpFor = function( type, textToHighlight ) {
+	const expressions = {};
 
-			return highlights;
-		}
+	expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
+	expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
+
+	return new RegExp( expressions[ type ], 'ig' );
+};
+
+const highlight = function( content, textToHighlight, type ) {
+	const matcher = getRegExpFor( type, textToHighlight ),
+		matches = matcher.exec( content );
```

Comment:
I respect you personal preference. :) However, saying "we're not using this indenting and comma usage with let and const and will need a seconds const here" is a bit strong if we don't have any mention of this in the [JS coding guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md). I couldn't find anything about this and even found things like the following:

![selection_189](https://cloud.githubusercontent.com/assets/4988512/19520178/9de13400-960f-11e6-8ac1-0fe3359a755d.png)

Btw, my personal preference is to do the same thing as @donnapep did here.